spec/forms/provider_interface/confirm_conditions_wizard_spec.rb

```diff
@@ -0,0 +1,120 @@
+require 'rails_helper'
+
+RSpec.describe ProviderInterface::ConfirmConditionsWizard do
+  let(:store) { instance_double(WizardStateStores::RedisStore) }
+  let(:conditions) { [build_stubbed(:offer_condition, status: :met), build_stubbed(:offer_condition)] }
+  let(:offer) { build_stubbed(:offer, conditions: conditions) }
+  let(:statuses) { nil }
+
+  let(:wizard) do
+    described_class.new(
+      store,
+      offer: offer,
+      statuses: statuses,
+    )
+  end
+
+  before { allow(store).to receive(:read) }
+
+  describe 'validations' do
+    let(:statuses) { { conditions.last.id.to_s => { 'status' => 'met' } } }
+
+    it 'validates that all conditions have a status set' do
+      expect(wizard).to be_invalid
+      expect(wizard.errors.first.message).to eq('Select a status')
```

Comment:
When I referred to the shoulda matches I thought this was part of the presence check of model, so ignore my comment. This looks good.